### PR TITLE
fix forwarding issues for packets destinated to AllThreadNodes when multiple SEDs exist

### PR DESCRIPTION
**Pull request by librasungirl**
@jwhui , please have a review.

### CODE REVIEW

File: src/core/thread/mesh_forwarder.cpp [hunk 1]
```diff
@@ -1014,6 +1014,7 @@ void MeshForwarder::HandleSentFrame(Mac::Frame &aFrame)
         {
             child->mFragmentOffset = 0;
             mSendMessage->ClearChildMask(mMle.GetChildIndex(*child));
+            mSendMessage->SetOffset(0);
```
Comment (jwhui) [inline src/core/thread/mesh_forwarder.cpp]: Thanks for catching t

**Comment by jwhui** [comment]
Thanks for catching this issue.  OpenThread maintains a fragment offset for each child in the child table.  I think a more complete fix would set the message offset based on the child's fragment offset when preparing to transmit.  What do you think about something like:

```c
void MeshForwarder::ScheduleTransmissionTask()
{
    ....
    for (int i = 0; i < numChildren; i++)
    {
     	if (children[i].mState == Child::kStateValid &&
            children[i].mDataRequest &&
            (mSendMessage = GetIndirectTransmission(children[i])) != NULL)
        {
            mSendMessage->SetOffset(children[i].mFragmentOffset);  ///< Set offset here.
            mMac.SendFrameRequest(mMacSender);
            ExitNow();
        }
    }
    ....
}
```

**Comment by librasungirl** [comment]
Good idea! Updated, thanks!

File: src/core/thread/mesh_forwarder.cpp [hunk 2]
```diff
@@ -190,6 +190,8 @@ void MeshForwarder::ScheduleTransmissionTask()
             children[i].mDataRequest &&
             (mSendMessage = GetIndirectTransmission(children[i])) != NULL)
         {
+            mSendMessage->SetOffset(children[i].mFragmentOffset);
+            mMac.SendFrameRequest(mMacSender);
```
Comment (jwhui) [inline src/core/thread/mesh_forwarder.cpp]: Looks like you duplicated a line here :)

Comment (librasungirl) [inline src/core/thread/mesh_forwarder.cpp]: kinda short... zzz
fixed

Comment (librasungirl) [inline src/core/thread/mesh_forwarder.cpp]: kinda short... zzz
fixed

**Comment by jwhui** [comment]
LGTM 👍

**Pull request merged**